static/js/country_details.js

```diff
@@ -0,0 +1,144 @@
+                        result[dataKey] = country[dataKey];
+                    }
+                });
+                return result;
+            });
+
+            return requestedData;
+        })
+        .catch(() => {
+            displayLoader(false);
+            maxPagesReached = true;
+            let errorMessageContainer = detailedCountryCard.querySelector('.error-message');
+
+            if (!errorMessageContainer) {
+                errorMessageContainer = document.createElement('div');
+                errorMessageContainer.classList.add('error-message');
+
+                errorMessageContainer.innerHTML = `
+                    <p>Something went wrong, please try again later...</p>
+                `;
+
+                detailedCountryCard.appendChild(errorMessageContainer);
+            }
+        });
+}
+
+const convertAbbrToCountryName = (abbr, data) => {
+    const convertedCountryNames = [];
+
+    abbr[0] === 'None' ? convertedCountryNames.push(abbr[0]) :
+        abbr.forEach((abbr) => {
+            const foundCountry = data.find(country => country["alpha3Code"] === abbr);
+
+            convertedCountryNames.push(foundCountry.name);
+        });
+
+    return convertedCountryNames;
+}
+
+const displayCard = (flag, name, nativeName, population, region, subregion, capital, topLevelDomain, currencies, languages, borders) => {
+    const card =
+        `<img class="detailed-flag" src="${flag}">
+    <div class="details-container">
+        <h2 class="detail-page-name">${name}</h2>
+
+        <ul class="details">
+            <li class="details-li"><strong>Native Name: </strong>${nativeName}</li>
+            <li class="details-li"><strong>Population: </strong>${population.toLocaleString('en-US')}</li>
+            <li class="details-li"><strong>Region: </strong>${region}</li>
+            <li class="details-li"><strong>Sub Region: </strong>${subregion}</li>
+            <li class="details-li"><strong>Capital: </strong>${capital}</li>
+        </ul>
+
+        <ul class="details">
+            <li class="details-li"><strong>Top Level Domain: </strong>${topLevelDomain}</li>
+            <li class="details-li"><strong>Currencies: </strong>${currencies.join(', ')}</li>
+            <li class="details-li"><strong>Languages: </strong>${languages.join(', ')}</li>
+        </ul>
+
+        <div class="border-countries-container">
+            <h3 class="border-countries-title"><strong>Border Countries:</strong></h3>
+            <div class="border-countries"></div>
+        </div>
+    </div>`;
+
+    return card
+}
+
+const displaySelectedCountry = () => {
+    const selectedCountry = localStorage.getItem('selected-country');
+
+    displayLoader(true)
+
+    fetchDetailedData(detailedInfo)
+        .then((data) => {
+            const countryToDisplay = data.find((country) => country.name === selectedCountry);
+            const currencies = [];
+            const languages = [];
+            let borderCountriesAbbr = [];
```

Comment:
waar staat `Abbr` voor?

Reply:
> waar staat `Abbr` voor?

abbreviation. Is het beter om het voluit te schrijven?

Reply:
Ja okay, hmm ja hoe je je variables wilt noemen is natuurlijk aan jou. Maar ik had wel een vermoeden dat het voor abbreviation staat. Maar ja, als andere naar je code kijken of gaan werken eraan, is het vaak handig om zo makkelijker voor ze te maken.